eq-author/src/App/sharing/SharePageContent/index.js

```diff
@@ -0,0 +1,155 @@
+import React from "react";
+import PropTypes from "prop-types";
+import config from "config";
+
+import { flowRight } from "lodash";
+import { Query, useMutation } from "react-apollo";
+
+import GET_QUESTIONNAIRE from "../graphql/GetQuestionnaire.graphql";
+import TOGGLE_PUBLIC_MUTATION from "../graphql/TogglePublicMutation.graphql";
+
+import Loading from "components/Loading";
+import Error from "components/Error";
+import ToggleSwitch from "components/buttons/ToggleSwitch";
+import { InformationPanel } from "components/Panel";
+import { withShowToast } from "components/Toasts";
+
+import EditorSearch from "./EditorSearch";
+
+import {
+  Layout,
+  PageTitle,
+  Description,
+  Section,
+  SectionTitle,
+  Separator,
+  FlexContainer,
+  PublicLabel,
+  ShareLinkButton,
+} from "../styles/SharePageContent";
+
+const propTypes = {
+  TogglePublicLabel: {
+    text: PropTypes.string.isRequired,
+    isActive: PropTypes.bool.isRequired,
+  },
+  Sharing: {
+    data: PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      isPublic: PropTypes.bool.isRequired,
+    }),
+    toast: PropTypes.func.isRequired,
+  },
+  GetQuestionnaireWrapper: {
+    questionnaireId: PropTypes.string,
+  },
+};
+
+export const TogglePublicLabel = ({ text, isActive }) => (
+  <PublicLabel isActive={isActive}>{text}</PublicLabel>
+);
+
+export const Sharing = ({ data, showToast }) => {
+  const { id, isPublic, createdBy, editors } = data.questionnaire;
+
+  const [updateIsPublic] = useMutation(TOGGLE_PUBLIC_MUTATION);
+
+  const previewUrl = `${config.REACT_APP_LAUNCH_URL}/${
+    (data.questionnaire || {}).id
+  }`;
+
+  const togglePublic = () =>
+    updateIsPublic({
+      variables: { input: { id, isPublic: !isPublic } },
+    });
+
+  const handleShareClick = () => {
+    const textField = document.createElement("textarea");
+    textField.setAttribute("data-test", "share-link");
+    textField.innerText = previewUrl;
+    document.body.appendChild(textField);
+    textField.select();
+    document.execCommand("copy");
+    textField.remove();
+    showToast("Link copied to clipboard");
+  };
+
+  return (
+    <Layout>
+      <PageTitle>Share your questionnaire</PageTitle>
+      <Description>
+        You can share your questionnaire with anyone who has an account with
+        Author.
+      </Description>
+      <ShareLinkButton
+        variant="tertiary"
+        small
+        onClick={handleShareClick}
+        data-test="share-button"
+      >
+        Get shareable link
+      </ShareLinkButton>
+      <Section>
+        <FlexContainer>
+          <SectionTitle>Public access</SectionTitle>
+          <Separator>
+            <TogglePublicLabel text="Off" isActive={!isPublic} />
```

Comment:
ToggleSwitch Now includes off/on within it - with a prop to activate it.
You could get rid of <TogglePublicLabel> if you wanted and just add  -  hideLabels={false} to ToggleSwitch to show the on/off labels.
This would bring it inline with other uses of ToggleSwitch as I think we are moving to add the on/off labels in more places.
Already implemented in settings page and settings modal on 'create new questionnaire'

(although the font size might be slightly bigger if you changed it)

Reply:
Nah I agree and think it's probably good to keep them all in line! I've swapped the component over now